### DISCUSSION TITLE
Thoughts on theme attribution

### DISCUSSION THREAD

**Discussion by jpanther** [jpanther/congo#192] [General]
One of the things that has started to become an issue is how Hugo and Congo are attributed in various places throughout the theme. As the theme author, I feel like there should be some attribution somewhere, but recognise that some people do not want this information so visible on every single page.

Like many themes, Hugo and Congo have always been attributed in the footer of the rendered page, and for some time this has been customisable using an i18n string that equates to "Powered by Hugo and Congo" with links to their respective pages. Prior to version 2.2, Hugo was also attributed in the HTML generator tag in the page `<head>`.

In version 2.2 this behaviour was adjusted to remove the generator tag from the HTML and this is now handled by the default Hugo logic. However, the question now remains how to best handle the attribution in the footer. Some people prefer to remove this altogether, while others want more customisation.

Since the very beginning, there has been an undocumented `attribution` parameter that allows the text to be removed, however it is a blunt instrument in that the CSS doesn't layout the page taking this into account, and it is better customised using the i18n string. As more people dig into the code and discover this 'feature', it's becoming something that people want to be able to have finer control over. I have received many requests to tweak this behaviour but as everyone has a different requirement, it's increasingly difficult to find a clear way forward. My current intention is to remove this parameter altogether, however I would like to replace it with something more useful.

Before making any further changes on this, I wanted to ask the community what your thoughts are on this topic in the hopes of coming to some sort of consensus on the best way to handle this.

The types of questions I'm trying to answer are:

- Do you mind having the theme attribution in the site footer?
- Do you want to only attribute Hugo, or Congo, or both?
- Do you disable the Hugo generator tag using the `disableHugoGeneratorInject` parameter?
- Do you currently hide the attribution using the `attribution` parameter, or some other way?
- Are you customising the i18n string to change the attribution text?
- Would you object to having the attribution displayed elsewhere (for example, in a HTML comment)?

_Related: https://github.com/jpanther/congo/issues/179, https://github.com/jpanther/congo/pull/191_

**Comment by AntoineSoetewey** [comment]
Dear James,

My thoughts on your questions:

- Do you mind having the theme attribution in the site footer?
Actually yes I do mind. I understand that you spent hundreds of hours building this amazing theme and you put a tremendous amount of work in updating and maintaining it. I will never thank you enough for this. However, I intentionally remove attributions to all my sites because I like my sites to be minimalist, so that it focuses on the most important information. If a visitor wants to know which theme I used, she/he can easily find it via GitHub or ask me (I'm be happy to redirect to the theme's repository). For my blog, I have been asked so many times that I have added the question in the FAQ (see [here](https://statsandr.com/faq/#3-what-technology-and-theme-do-you-use-to-write-this-blog-and-the-articles))
- Do you want to only attribute Hugo, or Congo, or both? If I can choose, none of them. It is not personal, I do it for all my sites and for all themes.
- Do you disable the Hugo generator tag using the disableHugoGeneratorInject parameter? Although I own several websites, I'm not an expert (I don't work in IT). I don't know what is the Hugo generator tag, so it's left with the default option.
- Do you currently hide the attribution using the attribution parameter, or some other way? Yes, using the parameter attribution = false.
- Are you customising the i18n string to change the attribution text? Again, not being an expert, I'm not sure what you're talking about, so this is left with the default option. All I can say is that in my websites there are no attribution and the copyright is usually either the name of the website, my name or both the name of the website and my name.
- Would you object to having the attribution displayed elsewhere (for example, in a HTML comment)? No, this is fine to me as long as it is not so visible and intrusive.
- last but not least: I acknowledge the time you must have given to build this theme and I really appreciate the effort you still put in it to update it (I hate to discover a great theme and see that the last commit was several years ago) and to answer people questions, so I have put a link to your website in my terms and policies page. See last section of this [page](https://antoinesoetewey.com/terms/). It is clear that most visitors will not see it, but at least Google knows about it. It is my way of thanking the theme's author while still have a minimalist website which suits my preference.

Hope this helps.

Antoine

**Reply by jpanther** [reply]
Thanks for the detailed reply! I actually think we have similar thoughts on this. I personally remove attribution when I use various themes too. However, I always post this information somewhere else as a way of thanking the original author.

**Comment by venteto** [comment]
I haven't gotten anywhere with this yet, but I'm trying to rework the footer partial to implement an optional long version of the footer attribution for both Hugo and Congo (and then create an additional configuration parameter, to not only enable footer attribution, but to also choose between a short and long variant (short being the existing implementation).

My desired long variant is like this:

Powered by [Hugo](https://gohugo.io/) ([0.101.0 extended](https://github.com/gohugoio/hugo/releases)) & [Congo](https://git.io/hugo-congo) ([2.3.0](https://github.com/jpanther/congo/blob/dev/CHANGELOG.md)). Site last updated: Sat 25 Jun 2022 16.51:32 UTC (1656129060).

I'd like to have the _version info_ separately link to each project's _changelogs_ so I can quickly get to them without having to go search for them (you will see 4 separate links in the above example).

I'm trying to use `hugo.Version`, `hugo.IsExtended`, and `hugo.Deps` to achieve this (and with `hugo,Deps`, regardless of whether or not I am using Hugo Modules, and selecting only the Congo element. I'm trying to make "extended" only print if it is true (e.g., it is true on my laptop but may not be true in the specific GitLab Pages image used).

In the event that the theme is (e.g., via Hugo Modules) is using a specific commit on the dev branch, then maybe the version info could be that specific, but it's not crucial.

Moreover, either on the same line as the attribution, or a separate line below it, I will try to implement an optional "Site last built" timestamp, both with the UTC time, formatted per the site's configuration, as well as an optional UNIX timestamp.

> Would you object to having the attribution displayed elsewhere (for example, in a HTML comment)?

Well, the only issue with that, for me, is that I try to eliminate all comments in production, so it would no longer be present, so the footer seems better for me personally.

**Reply by rshmhrj** [reply]
+1 for having the version numbers (maybe even behind a `hugo.IsProduction` flag so that we can see it only on dev)

Personally:

- Do you mind having the theme attribution in the site footer?
  - No problem from me
- Do you want to only attribute Hugo, or Congo, or both?
  - Attribute both
- Do you disable the Hugo generator tag using the disableHugoGeneratorInject parameter?
  - No -- have not modified this from your defaults
- Do you currently hide the attribution using the attribution parameter, or some other way?
  - No -- attribution is not hidden
- Are you customising the i18n string to change the attribution text?
  - No -- have not made any updates to the defaults
- Would you object to having the attribution displayed elsewhere (for example, in a HTML comment)?
  - No objections

Thinking about it more, for any other theme I would probably remove the attribution.  I think I have left it in @jpanther because I really like this project, and admire your dedication, brilliance, and your support to the community, and want to help you in whatever small ways I can.

**Comment by MikiVanousek** [comment]
I like to have the attribution to your theme on the bottom of my website. It allowed me to create an amazing website very easily, and it makes it clear to an observant visitor they can easily do the same thing. If someone wants to remove it, they could modify the footer partial, right?

I don't mind the mention of Hugo, but it might take some of the attention (about half) from Congo. I think the Congo attribution is more relevant, and the very first sentence of the GitHub README has the link to Hugo in it.

I do think it would be better if the link led to a landing page in the docs that has similar content as the current README. I think it's better branding. I also use the link to go to the docs of the theme regularly, and GitHub makes that into a journey twice as many clicks long.

**Comment by varnerac** [comment]
How about an option to pay/donate to remove attribution? I’ve done this with other themes.

**Reply by jpanther** [reply]
Thanks for the suggestion, however I have no intention to monetise the theme. Some people choose to [sponsor me on GitHub](https://github.com/sponsors/jpanther) as a way of saying thanks, and that's enough incentive to keep me working on new open source contributions.
